### PR TITLE
DAOS-2542 security: Check errors when making token

### PR DESCRIPTION
**Pull request by kjacque**
Errors converting uid/gid to names should not actually
be possible in practice, since the values are the effective
uid and gid. But this is a good practice anyway.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

**Comment by kjacque** [comment]
This PR depends on #475. The base branch will need to be changed to master when #475 is landed.

**Comment by kjacque** [comment]
> This PR depends on #475. The base branch will need to be changed to master when #475 is landed.

Changed the base and merged in master. Probably should have rebased instead - now the commit log is a little confusing.

Gatekeeper: Please use the PR title/description as commit message.

File: src/control/security/auth_sys.go [hunk 1]
```diff
@@ -108,17 +122,15 @@ func AuthSysRequestFromCreds(ext UserExt, creds *DomainInfo) (*auth.Credential,
 	// Marshal our AuthSys token into a byte array
 	tokenBytes, err := proto.Marshal(&sys)
 	if err != nil {
-		fmt.Errorf("Unable to marshal AuthSys token (%s)", err)
-		return nil, err
+		return nil, fmt.Errorf("Unable to marshal AuthSys token (%s)", err)
```
Comment (tanabarr) [inline src/control/security/auth_sys.go]: could use errors.WithMessagef here

Comment (kjacque) [reply src/control/security/auth_sys.go]: I think a few of the security files still need to be converted to use the errors package. Is there already a ticket? I can address it in a new patch.

Comment (kjacque) [reply src/control/security/auth_sys.go]: Couldn't find an existing ticket, so I created one: https://jira.hpdd.intel.com/browse/DAOS-2560

**Pull request merged**